src/state/mob.js

```diff
@@ -169,7 +170,16 @@ export function createMobModule(mobType) {
 				// Use ammo
 				var pocket = rootGetters["inventory/equipment"].pocket;
 				if (state.mobType == "player" && pocket.itemId && !rootGetters["inventory/checkRestricted"](pocket.itemId)) {
-					pocket.count -= 1;
+					if(rootGetters["upgrades/get"]("ammoSaver") > 0 && rootGetters["combat/isRanged"]) {
+						state.ammoStreak += 1;
+						if(state.ammoStreak == 5){
+							state.ammoStreak = 0;

```

Comment:
I'm less sure of a toast here since this happens more frequently, but it's worth considering. I'd be fine either way
```suggestion
							state.ammoStreak = 0;
							EventBus.$emit("toast", { icon: ITEMS[pocket.itemId].icon, text: `Ammo preserved!` });

```

Reply:
I think I would rather not, since this would get really spammy with faster guns (especially with chronosphere + the HP loss toasts)